docs/tutorials/basic/symbol.md

```diff
@@ -56,60 +111,150 @@ net = mx.sym.SoftmaxOutput(data=net, name='out')
 mx.viz.plot_network(net, shape={'data':(100,200)})
 ```
 
-### Modulelized Construction for Deep Networks
+Each symbol takes a (unique) string name.
+*Variable* typically represents inputs to the computation graph.
+Symbols defined via operators typically take other symbols as their input data,
+and might additionally accept other hyperparameters,
+such as the number of hidden neurons (*num_hidden*) or the activation type (*act_type*).
+
+We can view a symbol simply as a function taking several arguments.
+And we can retrieve those arguments with the following method call:
+
+```python
+net.list_arguments()
+```
+
+These arguments are the parameters needed by each symbol:
+
+- *data*: Input data needed by the variable *data*.
+- *fc1_weight* and *fc1_bias*: The weight and bias for the first fully connected layer *fc1*.
+- *fc2_weight* and *fc2_bias*: The weight and bias for the second fully connected layer *fc2*.
+- *out_label*: The label needed by the loss.
+
+We can also specify the automatically generated names explicitly:
+
+```python
+net = mx.symbol.Variable('data')
+w = mx.symbol.Variable('myweight')
+net = mx.symbol.FullyConnected(data=net, weight=w, name='fc1', num_hidden=128)
+net.list_arguments()
+```
+
+## More Complicated Composition
+
+MXNet provides well-optimized symbols for layers commonly used in deep learning
+(see [src/operator](https://github.com/dmlc/mxnet/tree/master/src/operator)).
+We can also define new operators in Python. The following example first
+performs an element-wise add between two symbols, then feeds them to the fully
+connected operator:
+
+```python
+lhs = mx.symbol.Variable('data1')
+rhs = mx.symbol.Variable('data2')
+net = mx.symbol.FullyConnected(data=lhs + rhs, name='fc1', num_hidden=128)
```

Comment:
explain somewhere that FullyConnected, for example, has 3 inputs, data, weight, bias. When you don't specify an input, an variable will be automatically generated for it.